test/record_test.cc

```diff
@@ -140,28 +140,27 @@ TEST_F(RecordFileTest, FirstTimeInit) {
   EXPECT_GT(ri_->get_free_space(ri_), 0xff);
 
   EXPECT_EQ(static_cast<size_t>(0), NumValidRecords());
-  EXPECT_NE(0, GetRecord(0, NULL));
+  EXPECT_NE(0, GetRecord(0, nullptr));
 }
 
 TEST_F(RecordFileTest, InitWithGarbage) {
   // Fill the file with garbage.
-  string garbage(4096, 0);
+  string garbage(4096, '\0');
   for (size_t i = 0; i < garbage.size(); ++i) {
     garbage[i] = i;
   }
-  FILE *file = fopen(filename_.c_str(), "w");
+  std::unique_ptr<FILE, int(*)(FILE *)> file(fopen(filename_.c_str(), "w"), fclose);
```

Comment:
should put this in a separate scope { } to guarantee that the file is closed/flushed before reading